src/ThemeJsonExtractor.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace WP_CLI\I18n;
+
+use Gettext\Extractors\Extractor;
+use Gettext\Extractors\ExtractorInterface;
+use Gettext\Translations;
+use WP_CLI;
+
+final class ThemeJsonExtractor extends Extractor implements ExtractorInterface {
+	use IterableCodeExtractor;
+
+	/**
+	 * @inheritdoc
+	 */
+	public static function fromString( $string, Translations $translations, array $options = [] ) {
```

Comment:
This function for the block.json extractor has a check to skip translation if the translation domains don't match.
https://github.com/dlind1/i18n-command/blob/d4a00c4e97007ffd8233b4ead7e869cf5c4226e1/src/BlockExtractor.php

Should we have a similar check? We don't deal with domains in this class, given that we are extracting strings from a theme are domains automatically handled?

Reply:
`theme.json` does not have a `domain` field, correct? In that case, the answer would be no.

Reply:
My understanding is that:

1. wp-cli already handles setting up the domain for the translation based on the type of input (in the case of themes, from the stylesheet).
2. while `block.json` provides an additional domain, `theme.json` doesn't, so we can't compare the `theme.json`'s domain to the one inferred by wp-cli.

Reply:
What Pascal said. I didn't see his comment until now (apparently, comments are only updated upon page reload).